launcher.py

```diff
@@ -12,94 +12,149 @@
 import webbrowser
 from importlib.metadata import PackageNotFoundError, version
 from pathlib import Path
+from typing import Optional
 
-CONFIG_FILE = Path("config.json")
-RUNTIME_CONFIG_FILE = Path("runtime_config.json")
-VENV_NAME = "virtual_env"  # For downloading python packages
-VENV_DIR = Path.cwd() / VENV_NAME  # Generate the path for the virtual environment
 MEDIAPROCESSOR_PATH = Path("MediaProcessor") / "build"
-DEFAULT_CONFIG = {
-    "version": 1,
-    "formatters": {
-        "detailed": {"format": "%(asctime)s [%(levelname)s] [%(funcName)s(), %(lineno)d]: %(message)s"},
-        "simple": {"format": "[%(levelname)s] %(message)s"},
-    },
-    "handlers": {
-        "console": {
-            "class": "logging.StreamHandler",
-            "formatter": "simple",
-            "stream": "ext://sys.stdout",
+LOG_LOCK = threading.Lock()
+
+
+class Config:
+    """
+    Manages Config files and logging Configuration.
+    """
+
+    CONFIG_FILE_PATH = Path("config.json")
+    RUNTIME_CONFIG_FILE_PATH = Path("runtime_config.json")
+    DEFAULT_LOGGING_CONFIG = {
+        "version": 1,
+        "formatters": {
+            "detailed": {"format": "%(asctime)s [%(levelname)s] [%(funcName)s(), %(lineno)d]: %(message)s"},
+            "simple": {"format": "[%(levelname)s] %(message)s"},
         },
-    },
-    "root": {"level": "ERROR", "handlers": ["console"]},
-}
+        "handlers": {
+            "console": {
+                "class": "logging.StreamHandler",
+                "formatter": "simple",
+                "stream": "ext://sys.stdout",
+            },
+        },
+        "root": {"level": "ERROR", "handlers": ["console"]},
+    }
 
+    def __init__(self, system: str, log_level: str, log_file: bool = False) -> None:
+        self.setup_runtime_config(system)
+        self.setup_logging(log_level, log_file)
 
-def setup_logging(log_level, log_file=False):
-    DEFAULT_CONFIG["root"]["level"] = log_level
-    if log_file:
-        DEFAULT_CONFIG["handlers"]["file"] = {
-            "class": "logging.handlers.RotatingFileHandler",
-            "formatter": "detailed",
-            "filename": f"logfile_{time.strftime('%Y%m%d_%H%M%S')}.log",
-            "maxBytes": 1024 * 1024 * 5,  # 5MB
-            "backupCount": 3,
-        }
-        DEFAULT_CONFIG["root"]["handlers"].append("file")
+    def setup_runtime_config(self, system: str):
+        try:
+            with open(Config.CONFIG_FILE_PATH, "r") as config_file:
+                config = json.load(config_file)
 
-    logging.config.dictConfig(DEFAULT_CONFIG)
+            if system == "Windows":
+                logging.info("Updating config file.")
+                for key, value in config.items():
+                    if type(value) == str:
+                        config[key] = value.replace("/", "\\")
```

Comment:
Please encapsulate this into `update_config_for_windows()`, as it makes it difficult to reason about `setup_runtime_config()` seeing a windows specific action without a comment, etc.